packages/x-date-pickers-pro/src/internal/hooks/useMultiInputRangeField/useMultiInputTimeRangeField.ts

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import useEventCallback from '@mui/utils/useEventCallback';
+import {
+  unstable_useTimeField as useTimeField,
+  UseTimeFieldComponentProps,
+} from '@mui/x-date-pickers/TimeField';
+import { useUtils, useValidation } from '@mui/x-date-pickers/internals';
+import { DateRange } from '../../models/range';
+import {
+  TimeRangeValidationError,
+  TimeRangeComponentValidationProps,
+  validateTimeRange,
+} from '../validation/useTimeRangeValidation';
+import {
+  UseMultiInputTimeRangeFieldDefaultizedProps,
+  UseMultiInputTimeRangeFieldParams,
+  UseMultiInputTimeRangeFieldProps,
+} from '../../../MultiInputTimeRangeField/MultiInputTimeRangeField.types';
+import { dateRangePickerValueManager } from '../../../DateRangePicker/shared';
+import { timeRangeFieldValueManager } from '../valueManager/timeRangeValueManager';
+
+export const useDefaultizedTimeRangeFieldProps = <TDate, AdditionalProps extends {}>(
+  props: UseMultiInputTimeRangeFieldProps<TDate>,
+): UseMultiInputTimeRangeFieldDefaultizedProps<TDate> & AdditionalProps => {
+  const utils = useUtils<TDate>();
+
+  const ampm = props.ampm ?? utils.is12HourCycleInCurrentLocale();
```

Comment:
I did not add the `ampm` prop to the `DateTimeField` and `TimeField`
If we are adding them today on the range fields we should add them there on another PR before the official release as well for consistency, and adapt the doc.

Reply:
Shouldn't that be left for locale/format to handle?
Major date libraries support different format for 12 or 24 hour clocks.
- [dayjs](https://day.js.org/docs/en/display/format#list-of-all-available-formats:~:text=0%2D23-,The%20hour,-HH)
- [moment](https://momentjs.com/docs/#/displaying/format/:~:text=am%20pm-,Hour,-H)
- [datefns](https://date-fns.org/v2.29.3/docs/format#:~:text=in%20the%20morning%2C%20...-,Hour%20%5B1%2D12%5D,-h)

Or was there a specific reason why we added separate `ampm` control? 🤔

Reply:
On the existing pickers you have a default which is based on the local.
The behavior is as follow:

```ts
const ampm = props.ampm ?? utils.is12HourCycleInCurrentLocale();
const inputFormat: ampm ? utils.formats.fullTime12h : utils.formats.fullTime24h,
```

So the `ampm` is just a shortcut to help user switch from 12h to 24h format without caring about the actual format.

Is this shortcut worth the overhead ? I'm not sure.